lib/core/configurator.c

```diff
@@ -693,6 +693,36 @@ static int on_config_unsetenv(h2o_configurator_command_t *cmd, h2o_configurator_
     return 0;
 }
 
+static int on_config_server_name(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
+{
+    char *server_name;
+    int ret;
+
+    ret = h2o_configurator_scanf(cmd, node, "%ms", &server_name);
```

Comment:
Together with https://github.com/h2o/h2o/pull/877/commits/75a3c723799dbf618f27ecbc1bc30ab2813756db#r60622942, you can simply use `h2o_strdup(NULL, node->data.scalar, SIZE_MAX)` as the server name.

And doing so makes me happy because I do not need to worry if support for the `m` specifier (introduced in POSIX.1-2008) is supported on various platforms :-)